<launch>
  <!-- Launch file to run bag file -->
  <arg name="bag_file_path" default="/home/renzo/catkin_ws/bag_mas_lento.bag"/>
  <arg name="folder_path" default="/home/renzo/catkin_ws"/>
  <arg name="post_procesamiento" default="true"/>
  <arg name="metodo" default="kmeans"/>

  <!-- Camera image left topic -->
  <arg name="left_camera_image_topic" default="/zed_lateral/zed_lateral/left/image_rect_color"/>
  <!-- Camera image right topic -->
  <arg name="right_camera_image_topic" default="/zed_lateral/zed_lateral/right/image_rect_color"/>
  <!-- Camera info left topic -->
  <arg name="left_camera_info_topic" default="/zed_lateral/zed_lateral/left/camera_info"/>
  <!-- Camera info right topic -->
  <arg name="right_camera_info_topic" default="/zed_lateral/zed_lateral/right/camera_info"/>

  <!-- Play the rosbag and remap camera info topics -->
  <node pkg="rosbag" type="play" name="rosbag_play" args="$(arg bag_file_path) $(arg left_camera_info_topic):=/stereo/left/camera_info $(arg right_camera_info_topic):=/stereo/right/camera_info" output="log"/>

  <!-- Remap left camera image topic -->
  <node pkg="image_transport" type="republish" name="remap_left_image" output="screen" args="compressed in:=$(arg left_camera_image_topic) raw out:=/stereo/left/image_raw" />
  <!-- Remap right camera image topic -->
  <node pkg="image_transport" type="republish" name="remap_right_image" output="screen" args="compressed in:=$(arg right_camera_image_topic) raw out:=/stereo/right/image_raw" />

  <!-- Run stereo_image_proc -->
  <group ns="stereo">
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" output="screen">
      <rosparam command="load" file="$(arg folder_path)/src/apple_fields_ignition/fields_ignition/config/stereoConfigMercedezv3.yaml"/>
    </node>
  </group>

  <!-- Run script to save the images -->
  <!-- use remap instead of the namespace -->
  <env name="ROS_NAMESPACE" value="stereo" />
  <node pkg="fields_ignition" type="save_disparity_and_rgb.py" name="save_disparity_and_rgb" args="--working_directory=$(arg folder_path) --post_procesamiento=$(arg post_procesamiento) --metodo=$(arg metodo)" output="screen"/>
</launch>

<!-- Ejemplo de ejecucion -->
<!-- Parado en la carpeta de catkin_ws -->
<!-- roslaunch fields_ignition stereo_sim_bag.launch bag_file_path:=/home/paolo/catkin_ws/s1_230228.bag folder_path:=/home/paolo/catkin_ws -->

<!--  rosbag play /home/renzo/catkin_ws/bag_mercedes_una_vuelta_completa.bag /zed_lateral/zed_lateral/left/camera_info:=/stereo/left/camera_info /zed_lateral/zed_lateral/right/camera_info:=/stereo/right/camera_info -->
<!--  rosrun image_transport republish compressed in:=/stereo/left/image_raw raw out:=/stereo/left/image_raw -->
<!--  rosrun image_transport republish compressed in:=/stereo/right/image_raw raw out:=/stereo/right/image_raw   -->
<!--  ROS_NAMESPACE=stereo rosrun stereo_image_proc stereo_image_proc -->
<!--  rosrun image_view stereo_view stereo:=/stereo image:=image_rect_color -->
<!--  rosrun rqt_reconfigure rqt_reconfigure -->
